src/theory/quantifiers_engine.cpp

```diff
@@ -854,10 +854,21 @@ void QuantifiersEngine::addTermToDatabase( Node n, bool withinQuant, bool within
     std::set< Node > added;
     d_term_db->addTerm(n, added, withinQuant, withinInstClosure);
 
-    //added contains also the Node that just have been asserted in this branch
-    if( d_quant_rel ){
-      for( std::set< Node >::iterator i=added.begin(), end=added.end(); i!=end; i++ ){
-        if( !withinQuant ){
+    if (!withinQuant)
+    {
+      if (d_sygus_tdb)
+      {
+        d_sygus_tdb->registerEvalTerm(n);
+      }
+
+      // added contains also the Node that just have been asserted in this
+      // branch
+      if (d_quant_rel)
+      {
+        for (std::set<Node>::iterator i = added.begin(), end = added.end();
+             i != end;
+             i++)
```

Comment:
It would be easier to read if you use a for each loop here.

Reply:
I prefer it this way, since it shows the type of added/i.

Reply:
I know it's a matter of preference, but in this case you could specify the type instead of `auto` and do e.g.

```
for (Node n : added)
{
  d_quant_rel->setRelevance(n.getOperator(), 0);
}
``` 
It's not relevant what `added` is in this loop. You can also use TNode for n in this case.

Reply:
I prefer having the type there so that if added changes types, I'd like the compiler to tell me to revisit this loop.
In my opinion, code should have a good level of redundancy to ensure that all changes are intentional, I don't like hiding details here.